### DISCUSSION TITLE
how to load a pickle in linux that is created/saved on windows machine?

### DISCUSSION THREAD

**Discussion by mahagilo** [explosion/spaCy#11475] [Help: Other Questions]
How can I load a pickle in Linux that is created/saved on a Windows machine?

I get the following error:

file "/usr/lib/python3.10/pathlib.py", line 960, in __new__
    raise NotImplementedError("cannot instantiate %r on your system"
NotImplementedError: cannot instantiate 'WindowsPath' on your system

**Comment by polm** [comment]
Can you clarify what it was you pickled? This might just not work. 

Also, is this related to spaCy specifically? If not this question might be better suited for Stack Overflow.

**Reply by mahagilo** [reply]
We built a keyword extractor and pickled it. We use spaCy. Any thoughts on how to fix this linux issue?

**Reply by polm** [reply]
I'm sorry, but with just that amount of detail I'm not sure what's going on. Can you please provide at least:

- sample code
- the full stack trace
- the pipeline config, if any

"Keyword extractor" sounds like an EntityRuler / PhraseMatcher, but it could be a couple of different things.

**Reply by mahagilo** [reply]
After a user types text in the GILO app’s editor, the keyword extractor API is triggered. The function of the keyword extractor API is to find and underline all keywords in a user's text. After this API is deployed, it knows the user's keywords, highlights them in a field-specific color, and can perform follow-up functions when the user clicks on a highlighted word.
Input:
text: Utf-8 or ASCII formatted text
Output:
JSON file with a list of extracted keywords. Every keyword is a tuple consisting of the following info:
•	Keyword in text: The correctly formatted keyword as found in the original text
•	Keyword in KnowledgeDB: The keyword is registered in the KnowledgeDB. This can vary from keywords in the text as single / plural, capitalization, or use of hyphens.
•	Field the keyword is in. 
•	Start and End byte of the keyword in the text.
•	RGB (Red, Green, Blue) color code for the underlining.
Sample of JSON file:
["Homicides", "homicide", "CRI", 4032, 4043, "45", "155", "66"]

![image](https://user-images.githubusercontent.com/80671503/189619778-b4e0ba97-1188-4ccc-80e6-44ddaf144c78.png)

**Reply by mahagilo** [reply]
PROBLEM SOLVED.

**Reply by polm** [reply]
Glad you figured it out. If you have any problems in the future, please include a sample of code that causes the issue and the complete stack trace.

**Reply by AadeshKabra** [reply]
Can you please post how this issue was resolved?
